sys/riscv/riscv/machdep.c

```diff
@@ -936,20 +936,28 @@ init_proc0(vm_offset_t kstack)
 static void
 try_load_dtb(caddr_t kmdp)
 {
-	vm_offset_t dtbp;
+	vm_ptr_t dtbp;
 
 	dtbp = MD_FETCH(kmdp, MODINFOMD_DTBP, vm_offset_t);
+#ifdef __CHERI_PURE_CAPABILITY__
+	if (dtbp != (vm_ptr_t)NULL) {
+		dtbp = (vm_ptr_t)cheri_andperm(cheri_setaddress(
+		    cheri_kall_capability, dtbp), CHERI_PERMS_KERNEL_DATA);
+		dtbp = (vm_ptr_t)cheri_setbounds((void *)dtbp,
+		    fdt_totalsize((void *)dtbp));
+	}
+#endif
 
 #if defined(FDT_DTB_STATIC)
 	/*
 	 * In case the device tree blob was not retrieved (from metadata) try
 	 * to use the statically embedded one.
 	 */
-	if (dtbp == (vm_offset_t)NULL)
-		dtbp = (vm_offset_t)&fdt_static_dtb;
+	if (dtbp == 0)
```

Comment:
```suggestion
	if (dtbp == (vm_ptr_t)NULL)
```
? Either way, this check and the one below should be made consistent.

Reply:
Done